src/notevennexus/json.py

```diff
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import json
+from typing import Any
+
+import numpy as np
+
+from .tree import Dataset, Group
+
+
+def read_json(path: str) -> Group:
+    """
+    Read JSON NeXus file and return tree of datasets and groups.
+
+    The JSON looks something like this:
+
+    {
+        "name": "delay",
+        "type": "group",
+        "attributes": [
+            {
+                "name": "NX_class",
+                "dtype": "string",
+                "values": "NXlog"
+            }
+        ],
+        "children": [
+            {
+                "module": "f142",
+                "config": {
+                    "source": "source",
+                    "topic": "topic",
+                    "dtype": "double",
+                    "value_units": "ns"
+                },
+                "attributes": [
+                    {
+                        "name": "units",
+                        "dtype": "string",
+                        "values": "ns"
+                    }
+                ]
+            },
+            {
+                "module": "dataset",
+                "config": {
+                    "name": "slits",
+                    "values": 1,
+                    "type": "int64"
+                }
+            },
+        ]
+    },
+    """
+    with open(path, "r") as f:
+        return _read_group(json.load(f))
+
+
+def _read_group(group: dict[str, Any], parent: Group | None = None) -> Group:
+    """Read JSON group"""
+    name = group.get("name", '')
+    if parent is not None:
+        name = parent.name + '/' + name
+    grp = Group(name=name, attrs={}, children={}, parent=parent)
+    for child in group["children"]:
+        if not isinstance(child, dict):
+            continue
+        module = child.get("module")
+        if module is None:
+            if child["type"] == "group":
+                grp.children[child["name"]] = _read_group(child, parent=grp)
+        elif module == "dataset":
+            grp.children[child["config"]["name"]] = _read_dataset(child, parent=grp)
+        elif module in ["f142", 'f144']:
+            grp.children[child["config"]["source"]] = _read_source(child, parent=grp)
+        elif module in ['tdct', 'ev44']:
+            # No useful info in these?
+            pass
+        else:
+            raise ValueError(f"Unsupported module: {module}")
```

Comment:
So this fails if it cannot read the group.
In the mechanism of validators, the bad things in the file are just reported in the results, instead of raising an error.
Should a similar mechanism be used here?

Reply:
It is not really the same, since the "modules" have nothing o do with NeXus, i.e., it is not what we are validating. We could skip it, maybe. I think the JSON bit will need more iterations anyway: I am not sure if there is anything standard about it, or if it is just some bespoke ESS format that can change any time?